test/spec_encryptor.rb

```diff
@@ -0,0 +1,51 @@
+require 'minitest/autorun'
+require 'rack/encryptor'
+require 'rack/utils'
+
+describe Rack::Encryptor do
+  def setup
+    @secret = OpenSSL::Cipher.new(Rack::Encryptor::ENCRYPTION_CIPHER).random_key
+  end
+
+  it 'encrypted message contains ciphertext iv and auth_tag' do
+    msg = Rack::Encryptor.encrypt_message('hello world', @secret)
+
+    ctxt, iv, auth_tag = msg.split(Rack::Encryptor::ENCRYPTION_DELIMITER, 3)
+
+    ctxt.wont_be :empty?
+    iv.wont_be :empty?
+    auth_tag.wont_be :empty?
+  end
+
+  it 'encrypted message is decryptable' do
+    cmsg = Rack::Encryptor.encrypt_message('hello world', @secret)
+    pmsg = Rack::Encryptor.decrypt_message(cmsg, @secret)
+
+    pmsg.must_equal 'hello world'
+  end
+
+  it 'encryptor and decryptor handles overly long keys' do
+    new_secret = "#{@secret}abcdef123456"
+
+    # These methos should truncate the long key (so OpenSSL raise exceptions)
```

Comment:
Typo: methos

Reply:
Q: does the last bit of this sentence mean "so as to avoid OpenSSL raising exceptions"?

Reply:
Thanks!